App/Controllers/RoleController.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+namespace App\Controllers;
+
+use Core\View;
+use App\Models\Role;
+use App\Models\Permission;
+
+class RoleController {
+
+  public function index()
+  {
+      $query = Role::getDB();
+      $roles = $query->readAll(Role::getTableName(),Role::class);
+
+      View::render("roles/index.view.php",["roles" => $roles]);
+  }
+
+  public function create()
+  {
+      View::render("roles/create.view.php");
+  }
+
+  public function edit($id)
+  {
+      if (isset($id))
```

Comment:
Maybe here in case of id not existing to return an error message to the user.

Reply:
Should I display the edit screen of an entity and show an error message that the entity was not found or is better to redirect to an entity not found page?

Reply:
show that entity not found.